[AnomalyDetection] Add a notebook for anomaly detection with Z-Score

shunping opened the pull request:
colab link:
https://colab.sandbox.google.com/github/shunping/beam/blob/anomaly-detection-notebook-1/examples/notebooks/beam-ml/anomaly_detection/anomaly_detection_zscore.ipynb

damccorm commented on examples/notebooks/beam-ml/anomaly_detection/anomaly_detection_zscore.ipynb:
```diff
@@ -0,0 +1,653 @@
+        "colab_type": "text"
+      },
+      "source": [
+        "<a href=\"https://colab.research.google.com/github/shunping/beam/blob/anomaly-detection-notebook-1/anomaly_detection_zscore.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
```
Please remove in favor of a cell below the title like https://github.com/apache/beam/blob/4ac20255c993906045bb720618324575b50689d7/examples/notebooks/beam-ml/automatic_model_refresh.ipynb#L45 (we do this across all our notebooks)

shunping replied:
Fixed.

damccorm commented on examples/notebooks/beam-ml/anomaly_detection/anomaly_detection_zscore.ipynb:
```diff
@@ -0,0 +1,653 @@
+    {
+      "cell_type": "code",
+      "source": [
+        "! pip install apache_beam[interactive,gcp]==2.64.0rc2 --quiet"
```
```suggestion
        "! pip install apache_beam[interactive,gcp]>=2.64.0 --quiet"
```

Adding this so we don't forget

damccorm commented on examples/notebooks/beam-ml/anomaly_detection/anomaly_detection_zscore.ipynb:
```diff
@@ -0,0 +1,653 @@
+        "id": "0064575d-5e60-4f8b-a970-9dc39db8d331"
+      },
+      "source": [
+        "## Generating Synthetic Data with Concept Drifting"
```
```suggestion
        "## Generating Synthetic Data with Concept Drift"
```

Nit: I think this is the right tense/may help with SEO

damccorm reviewed (commented):
Only had minor feedback, overall this looks great! We should wait until 2.64.0 is released before merging, but it should be ready to go

damccorm commented:
waiting on author

shunping commented:
Thanks for approving. I will need to update the beam version to a released version.

claudevdm commented:
Should we add links like https://github.com/apache/beam/blob/891e8a447042b717f194db3120d533ae2644446f/examples/notebooks/beam-ml/custom_remote_inference.ipynb#L42?

shunping commented:
> Should we add links like https://github.com/apache/beam/blob/891e8a447042b717f194db3120d533ae2644446f/examples/notebooks/beam-ml/custom_remote_inference.ipynb#L42?

I think @damccorm mentioned those links are automatically added.

damccorm commented:
> > Should we add links like https://github.com/apache/beam/blob/891e8a447042b717f194db3120d533ae2644446f/examples/notebooks/beam-ml/custom_remote_inference.ipynb#L42
> > ?
> 
> I think @damccorm mentioned those links are automatically added.

No, we still need to add them - my feedback was just about the formatting

shunping commented:
@damccorm, could you please take another look?

Some changes since your last review:
- Added a new section for running anomaly detection on a batch data set.
- Added the links for opening the notebook in colab and github.
- Polish the intro
- Add a loop in the drawing at the end to show results as more data streaming in.

damccorm reviewed (approved):
Thanks!
